edit-post/store/utils.js

```diff
@@ -2,12 +2,16 @@
  * Given a selector returns a functions that returns the listener only
  * if the returned value from the selector changes.
  *
- * @param  {function} selector Selector.
- * @param  {function} listener Listener.
- * @return {function}          Listener creator.
+ * @param  {function} selector       Selector.
+ * @param  {function} listener       Listener.
+ * @param  {boolean}  executeAtStart true if listener should be executed during the onChangeListener creation.
+ * @return {function}                Listener creator.
  */
-export const onChangeListener = ( selector, listener ) => {
+export const onChangeListener = ( selector, listener, executeAtStart = false ) => {
```

Comment:
I'm not sure about this param. It doesn't line up with e.g. how [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) works which is what I imagine most readers would expect.

Could we leave it up to the caller?

```jsx
function adjustSidebar() {
	...
}
onChangeListener(
	() => select( 'core/viewport' ).isViewportMatch( '< medium' ),
	adjustSidebar
);
adjustSidebar();
```

Reply:
Thank you for the feedback @noisysocks it was applied.